WcaOnRails/app/models/competition.rb

```diff
@@ -622,6 +622,62 @@ def continent
     country ? Continent.find_by_id(country.continentId) : nil
   end
 
+  def psych_sheet_event(event)
+    preferred_format = event.preferred_formats.first
+    sort_clause = (preferred_format.sort_by == :single) ? "-single_rank desc, -average_rank desc, users.name" : "-average_rank desc, -single_rank desc, users.name"
+
+    joinsql = <<-ENDSQL
+      join users on users.id = Preregs.user_id
+      join Countries on Countries.iso2 = users.country_iso2
+      left join RanksSingle on RanksSingle.personId = users.wca_id and RanksSingle.eventId = '#{event.id}'
+      left join RanksAverage on RanksAverage.personId = users.wca_id and RanksAverage.eventId = '#{event.id}'
+    ENDSQL
+
+    selectsql = <<-ENDSQL
+      Preregs.id,
+      users.name select_name,
+      users.wca_id select_wca_id,
+      Preregs.accepted_at,
+      Countries.name select_country,
+      registration_events.event_id,
+      RanksAverage.worldRank average_rank,
+      ifnull(RanksAverage.best, 0) average_best,
+      RanksSingle.worldRank single_rank,
+      ifnull(RanksSingle.best, 0) single_best
+    ENDSQL
+
+    registrations = self.registrations.
+                         accepted.
+                         joins(joinsql).
+                         joins(:registration_events).
+                         where("registration_events.event_id=?", event.id).
```

Comment:
@jonatanklosko it doesn't appear to work as it brings back other events besides 333 when this is only specified in the join condition. Here's the SQL that doesn't work:

`SELECT       Preregs.id,       users.name select_name,       users.wca_id select_wca_id,       Preregs.accepted_at,       Countries.name select_country,       registration_events.event_id,       RanksAverage.worldRank average_rank,       ifnull(RanksAverage.best, 0) average_best,       RanksSingle.worldRank single_rank,       ifnull(RanksSingle.best, 0) single_best  FROM `Preregs` join registration_events on registration_events.registration_id = Preregs.id and registration_events.event_id = 333       join users on users.id = Preregs.user_id       join Countries on Countries.iso2 = users.country_iso2       left join RanksSingle on RanksSingle.personId = users.wca_id and RanksSingle.eventId = '333'       left join RanksAverage on RanksAverage.personId = users.wca_id and RanksAverage.eventId = '333' WHERE `Preregs`.`competitionId` = 'Euro2016' AND (`Preregs`.`accepted_at` IS NOT NULL)  ORDER BY -average_rank desc, -single_rank desc, users.name;`

If the condition is in the where clause you get the right result:

`SELECT       Preregs.id,       users.name select_name,       users.wca_id select_wca_id,       Preregs.accepted_at,       Countries.name select_country,       registration_events.event_id,       RanksAverage.worldRank average_rank,       ifnull(RanksAverage.best, 0) average_best,       RanksSingle.worldRank single_rank,       ifnull(RanksSingle.best, 0) single_best  FROM `Preregs` join registration_events on registration_events.registration_id = Preregs.id       join users on users.id = Preregs.user_id       join Countries on Countries.iso2 = users.country_iso2       left join RanksSingle on RanksSingle.personId = users.wca_id and RanksSingle.eventId = '333'       left join RanksAverage on RanksAverage.personId = users.wca_id and RanksAverage.eventId = '333' WHERE `Preregs`.`competitionId` = 'Euro2016' AND (`Preregs`.`accepted_at` IS NOT NULL) and registration_events.event_id = '333'  ORDER BY -average_rank desc, -single_rank desc, users.name;
`

I don't understand why this is happening. If I simplify it for 1 user it looks like it's bringing back any event beginning with 333. I think this is a bug but I'll need to revert to the filter being in the where clause.

`SELECT * FROM `Preregs` join registration_events on registration_events.registration_id = Preregs.id and registration_events.event_id = 333 where competitionId = 'Euro2016' and user_id = 7831`